test/unit/Cardano/Wallet/BlockSyncerSpec.hs

```diff
@@ -118,11 +115,8 @@ instance Arbitrary Blocks where
     arbitrary = do
         n <- fromIntegral . (`mod` 42) <$> arbitrary @Word8
         let h0 = BlockHeader (SlotId 1 0) (Hash "initial block")
-        let blocks = map snd $ take n $ iterate next
-                ( blockHeaderHash h0
-                , Block h0 mempty
-                )
-        mapM duplicateMaybe blocks >>= fmap Blocks . groups . mconcat
+        let b0 = (blockHeaderHash h0, Block h0 mempty)
+        Blocks <$> groups (map snd $ take n $ iterate next b0)
```

Comment:
does it going to produce duplicated Blocks ?

Reply:
Nope it's not. I've removed that responsibility from the ticking function for now, and we'll have to make this a responsibility of the network layer.